File: src/main/java/com/blogly/blogly/infrastructure/security/DelegatingPasswordHasher.java
package com.blogly.blogly.infrastructure.security;

import com.blogly.blogly.domain.user.PasswordHasher;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class DelegatingPasswordHasher implements PasswordHasher {

    private final PasswordEncoder passwordEncoder;

    @Override
    public String hash(String plainText) {
        return passwordEncoder.encode(plainText);
    }

    @Override
    public boolean matches(String plainText, String hashedValue) {
        return passwordEncoder.matches(plainText, hashedValue);
    }
}

File: src/main/java/com/blogly/blogly/infrastructure/security/DelegatingPasswordHasher.kt
package com.blogly.blogly.infrastructure.security

import com.blogly.blogly.domain.user.PasswordHasher
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Component

@Component
class DelegatingPasswordHasher(private val passwordEncoder: PasswordEncoder) : PasswordHasher {

    override fun hash(plainText: String) = passwordEncoder.encode(plainText)!!

    override fun matches(plainText: String, hashedValue: String) =
        passwordEncoder.matches(plainText, hashedValue)
}
